fuse/dl/cross_validation/pipeline.py

```diff
@@ -0,0 +1,303 @@
+from typing import Dict, Callable
+from fuse.utils import gpu as FuseUtilsGPU
+from sklearn.model_selection import KFold
+from functools import partial
+from multiprocessing import Process, Queue
+from typing import Sequence, Union
+import os
+from fuse.eval.metrics.classification.metrics_ensembling_common import MetricEnsemble
+from collections import OrderedDict
+from fuse.eval.evaluator import EvaluatorDefault
+from fuse.utils.file_io.file_io import create_or_reset_dir
+from fuse.utils.rand.seed import Seed
+
+
+def ensemble(test_dirs: Sequence[str], test_infer_filename: str, target_key: str, ensembled_output_file: str) -> None:
+    ensembled_output_dir = os.path.dirname(ensembled_output_file)
+    create_or_reset_dir(ensembled_output_dir, force_reset=True)
+    test_infer_filenames = [os.path.join(d, test_infer_filename) for d in test_dirs]
+    # define data for ensemble metric
+    data = {str(k): test_infer_filenames[k] for k in range(len(test_infer_filenames))}
+
+    # list of metrics
+    metrics = OrderedDict(
+        [
+            (
+                "ensemble",
+                MetricEnsemble(
+                    pred_keys=[str(i) + ".model.output.classification" for i in range(len(test_dirs))],
```

Comment:
You assume here that the prediction stored in model.output.classification.
You should get it as an input.
We might have multiple keys we would like ensemble - keep it in mind.

Reply:
added it as input